### ISSUE TITLE
[BUG] Error Code 139 when Exception is Thrown in Method Call with Span

### ISSUE THREAD

**Issue by Bibob7** [DataDog/dd-trace-php#879]
**Describe the bug**
If there is an exception thrown in a method that I define a span for with `dd_trace_method`, I see error code 139 or Segmentation Fault and the process crashes. 

**PHP Info**
- Output php -v:
```
PHP 7.4.3 (cli) (built: Feb 20 2020 21:45:11) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with ddtrace v0.44.1, Copyright Datadog, by Datadog
    with blackfire v1.30.0~linux-musl-x64-non_zts74, https://blackfire.io, by Blackfire
```
-Output `php -m`

```[PHP Modules]
blackfire
Core
ctype
curl
date
ddtrace
dom
fileinfo
filter
ftp
grpc
hash
iconv
json
libxml
mbstring
mongodb
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
pdo_sqlite
Phar
posix
protobuf
rdkafka
readline
Reflection
session
SimpleXML
sodium
SPL
sqlite3
standard
tokenizer
xml
xmlreader
xmlwriter
zlib

[Zend Modules]
blackfire
ddtrace
```

**OS Info**
- Output of `cat /etc/os-release | grep -E "(NAME)|(VERSION)"`
```
NAME="Alpine Linux"
VERSION_ID=3.11.3
PRETTY_NAME="Alpine Linux v3.11"
```
**Version**
- dd-trace-php version 0.44.1

**Comment by SammyK** [comment]
Hey @Bibob7. Thank you for reporting this issue. We're trying to reproduce this issue on our end. Would you be able to provide the snippet of code with the `dd_trace_method()` call?

**Comment by Bibob7** [comment]
Hey @SammyK, it seems only to happen, if the exception is thrown in a sub method call. Here a simple example:

```
<?php
dd_trace_method(
    TestClass::class,
    'method',
    function (\DDTrace\SpanData $span, array $args) {
        $span->name = 'test.exception.in.sub.method';
    }
);

$withLoop = new TestClass();
$withLoop->method();

class TestClass {
    /**
     * @throws Exception
     */
    public function method(): void
    {
        $this->subMethod();
    }

    /**
     * @throws Exception
     */
    private function subMethod(): void
    {
        throw new Exception('Test');
    }
}
```

**Comment by SammyK** [comment]
Thank you so much for this extra info @Bibob7! I can confirm that we are able to replicate this issue on our end and are woking on a fix for it now. I will follow up with you as soon as we have a patch ready.

**Comment by SammyK** [comment]
Hey @Bibob7! A fix for this was released in [0.45.1](https://github.com/DataDog/dd-trace-php/releases/tag/0.45.1). Would you mind confirming that this patch fixes the issue for you?

**Comment by labbati** [comment]
Hi @Bibob7 I am closing this issue as we were able to reproduce it and fix in our latest release.
Thanks for taking the time to report it.
